WordPress/Classes/Services/NotificationSyncService.swift

```diff
@@ -0,0 +1,304 @@
+import Foundation
+
+
+// MARK: - Notifications
+//
+let NotificationSyncServiceDidUpdateNotifications = "NotificationSyncServiceDidUpdateNotifications"
+
+
+// MARK: - NotificationSyncService
+//
+class NotificationSyncService
+{
+    /// Returns the Main Managed Context
+    ///
+    private var contextManager = ContextManager.sharedInstance()
+
+    /// Sync Service Remote
+    ///
+    private var remote: NotificationSyncServiceRemote!
+
+    /// Maximum number of Notes to Sync
+    ///
+    private let maximumNotes = 100
+
+
+    /// Designed Initializer
+    ///
+    init?() {
+        guard let dotcomAPI = dotcomAPI else {
+            return nil
+        }
+
+        remote = NotificationSyncServiceRemote(wordPressComRestApi: dotcomAPI)
+    }
+
+    /// Initializer: Useful for Unit Testing
+    ///
+    /// - Parameters:
+    ///     - manager: ContextManager Instance
+    ///     - wordPressComRestApi: The WordPressComRestApi that should be used.
+    ///
+    init?(manager: ContextManager, dotcomAPI: WordPressComRestApi) {
+        guard dotcomAPI.hasCredentials() else {
+            return nil
+        }
+
+        contextManager = manager
+        remote = NotificationSyncServiceRemote(wordPressComRestApi: dotcomAPI)
+    }
+
+
+
+    /// Syncs the latest *maximumNotes*:
+    ///
+    /// - Latest 100 hashes are retrieved (++efficiency++)
+    /// - Only those Notifications that were remotely changed (Updated / Inserted) will be retrieved
+    /// - Local collection will be updated. Old notes will be purged!
+    ///
+    func sync(completion: ((ErrorType?, Bool) -> Void)? = nil) {
+        assert(NSThread.isMainThread())
+
+        remote.loadHashes(withPageSize: maximumNotes) { error, remoteHashes in
+            guard let remoteHashes = remoteHashes else {
+                completion?(error, false)
+                return
+            }
+
+            self.deleteLocalMissingNotes(from: remoteHashes) {
```

Comment:
Thanks @frosty, will iterate in another PR. We're capping the number of notifications to 100, and working in BG, but agreed, let's revisit this.